### DISCUSSION TITLE
how to give two datasets containing different elements in dpgen

### DISCUSSION THREAD

**Discussion by kyuan2019** [deepmodeling/dpgen#1296] [Q&A]
Dear all,
I have two training data but they contain different elements. Here is my input file for the beginning part of the param.jason:
{
     "type_map": ["Mg","O","H","Na","C"],
     "init_data_prefix": "../",
     "init_data_sys":[
             ["init/mgo_NaHCO3_330K/training_data"], #notes: this file contains all the type
             ["init/mgo444_330K/training_data"] #notes: this one only contains Mg and O, and I also checked the type.raw file to make sure Mg is 0 and O is 1.
     ],
     "sys_configs_prefix": "../",
     "sys_configs": [
         ["init/mgo_NaHCO3_330K/POSCAR_FILES/00001/POSCAR"]
     ],
     "_comment": " that's all ",
     "numb_models": 1,
     "train_param": "input.json",
     "default_training_param": {
         "model": {
             "type_map": ["Mg","O","H","Na","C"], # notes: I put every element here
             "descriptor": {
                 "type": "se_e2_a",
                 "sel": [16,95,95,95,95],
                 "rcut_smth": 0.5,
                 "rcut": 6.0,
                 "neuron": [25,50,100],
                 "resnet_dt": true,
                 "axis_neuron": 12,
                 "seed": 5678
             },
             "fitting_net": {
                 "neuron": [120,120,120],
                 "resnet_dt": true,
                 "seed": 5678
             }
         },

After I run the code, it shows an error:

nohup: ignoring input
Traceback (most recent call last):
  File "/home/cloud/anaconda3/envs/deepmd/bin/dpgen", line 8, in <module>
    sys.exit(main())
  File "/home/cloud/anaconda3/envs/deepmd/lib/python3.10/site-packages/dpgen/main.py", line 233, in main
    args.func(args)
  File "/home/cloud/anaconda3/envs/deepmd/lib/python3.10/site-packages/dpgen/generator/run.py", line 5109, in gen_run
    run_iter(args.PARAM, args.MACHINE)
  File "/home/cloud/anaconda3/envs/deepmd/lib/python3.10/site-packages/dpgen/generator/run.py", line 4437, in run_iter
    make_train(ii, jdata, mdata)
  File "/home/cloud/anaconda3/envs/deepmd/lib/python3.10/site-packages/dpgen/generator/run.py", line 353, in make_train
    sys_paths = expand_sys_str(os.path.join(init_data_prefix, ii))
  File "/home/cloud/anaconda3/envs/deepmd/lib/python3.10/posixpath.py", line 90, in join
    genericpath._check_arg_types('join', a, *p)
  File "/home/cloud/anaconda3/envs/deepmd/lib/python3.10/genericpath.py", line 152, in _check_arg_types
    raise TypeError(f'{funcname}() argument must be str, bytes, or '
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'list'

Any idea on how to solve this issue? Thanks!

**Comment by njzjz** [comment]
`init_data_sys` should be an 1D list.

**Reply by kyuan2019** [reply]
Thanks for your reply. What do you mean by a "1D" list? Here is my init_data_sys

"init_data_sys":[
["init/mgo_NaHCO3_330K/training_data"], #Contains Mg, O, H, Na, C 
["init/mgo444_330K/training_data"] # contains only Mg and O

So you mean both data should have the same types of elements? In fact, the first one is a MgO surface in NaHCO3 solution and the second one is just a mgo bulk crystal. Is there a way to include both in the dpgen init_data_sys?

Thanks!

**Reply by Tong-Zhou223** [reply]
> Thanks for your reply. What do you mean by a "1D" list? Here is my init_data_sys
> 
> "init_data_sys":[ ["init/mgo_NaHCO3_330K/training_data"], #Contains Mg, O, H, Na, C ["init/mgo444_330K/training_data"] # contains only Mg and O
> 
> So you mean both data should have the same types of elements? In fact, the first one is a MgO surface in NaHCO3 solution and the second one is just a mgo bulk crystal. Is there a way to include both in the dpgen init_data_sys?
> 
> Thanks!

Excuse me, do you know how to write this order of the elements in param.json file now?
